src/NuGetForUnity/Editor/NugetPackageIdentifier.cs

```diff
@@ -345,127 +403,202 @@ public int CompareVersion(string otherVersion)
                 if (IsMaxInclusive)
                 {
                     // if there is no MaxVersion specified, but the Max is Inclusive, then it is an EXACT version match with the stored MINIMUM
-                    return CompareVersions(MinimumVersion, otherVersion);
+                    return compareMinimum;
                 }
             }
 
             return 0;
         }
 
         /// <summary>
-        ///     Compares two version numbers in the form "1.2". Also supports an optional 3rd and 4th number as well as a prerelease tag, such as
-        ///     "1.3.0.1-alpha2".
-        ///     Returns:
-        ///     -1 if versionA is less than versionB
-        ///     0 if versionA is equal to versionB
-        ///     +1 if versionA is greater than versionB
+        ///     SemVer2 <see href="https://semver.org/" />.
+        ///     Complying with NuGet comparison rules <see href="https://learn.microsoft.com/en-us/nuget/concepts/package-versioning" />.
         /// </summary>
-        /// <param name="versionA">The first version number to compare.</param>
-        /// <param name="versionB">The second version number to compare.</param>
-        /// <returns>-1 if versionA is less than versionB. 0 if versionA is equal to versionB. +1 if versionA is greater than versionB</returns>
-        private static int CompareVersions(string versionA, string versionB)
+        private readonly struct SemVer2Version
         {
-            try
+            private readonly int major;
+
+            private readonly int minor;
+
+            private readonly int? patch;
+
+            private readonly int? build;
+
+            private readonly string buildMetadata;
+
+            public string PreRelease { get; }
+
+            /// <summary>
+            ///     Initializes a new instance of the <see cref="SemVer2Version" /> struct.
+            /// </summary>
+            /// <param name="dummy">Dummy to allow calling this constructor.</param>
+            public SemVer2Version(bool dummy)
```

Comment:
Maybe a better option would be a `static SemVer2Version Create()` method?

Reply:
I never used `struct` in a old C# language version where there Default constructor is not supported.
I just didn't hat that in mind. If you like you can change it.